### ISSUE TITLE
ares_init.c build fails with 'incompatible pointer type' under mingw on windows.

### ISSUE THREAD

**Issue by jtattermusch** [c-ares/c-ares#143]
function next_suffix asks for `const char**`  but pp is just `char**`

```
third_party/cares/cares/ares_init.c: In function 'get_SuffixList_Windows':
third_party/cares/cares/ares_init.c:1430:7: error: passing argument 1 of 'next_suffix' from incompatible pointer type [-Werror]
       while (len = next_suffix(&pp, len))
       ^
third_party/cares/cares/ares_init.c:1347:15: note: expected 'const char **' but argument is of type 'char **'
 static size_t next_suffix(const char** list, const size_t advance)
```

**Comment by jtattermusch** [comment]
CC @y-zeng

**Comment by bradh352** [comment]
fixed in 2effec2
